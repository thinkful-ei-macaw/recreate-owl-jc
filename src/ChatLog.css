.chat-box {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  
}