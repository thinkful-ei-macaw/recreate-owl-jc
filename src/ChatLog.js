import React from 'react'
import ChatEvent from './ChatEvent';
import {findNameById} from './participantDummy';

const ChatLog = (props) => {


return (
<div>
  {props.chatEvents.map(event => (
        <ChatEvent 
          key={event.participantId}
          name={findNameById(event.participantId)}
          type={event.type}
          message={event.message}
          time={event.time}
          timestamp={event.timestamp}
  />))}
</div>
);
}

export default ChatLog