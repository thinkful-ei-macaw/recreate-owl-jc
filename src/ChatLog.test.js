import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './ChatLog';
import renderer from 'react-test-renderer';

describe('ChatLog Component tests', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
      ReactDOM.render(<ChatLog chatEvents={[{
        participantId: 3,
        type: 'message',
        message: 'Hello world',
        time: 1548852646559,
        timestamp: 1548852484247
      },
      {
        participantId: 2,
        type: 'thumbs-up',
        timestamp: 1548852484247
      },
      {
        participantId: 4,
        type: 'thumbs-down',
        timestamp: 1548852484247
      }]} />, div);
      ReactDOM.unmountComponentAtNode(div);
    
    
    });

    it('renders the expected events in log', () => {
      const tree = renderer
      .create(<ChatLog chatEvents={[{
        participantId: 3,
        type: 'message',
        message: 'Hello world',
        time: 1548852646559,
        timestamp: 1548852484247
      },
      {
        participantId: 2,
        type: 'thumbs-up',
        timestamp: 1548852484247
      },
      {
        participantId: 4,
        type: 'thumbs-down',
        timestamp: 1548852484247
      }]} />)
      .toJSON();
      expect(tree).toMatchSnapshot();
    });

});

