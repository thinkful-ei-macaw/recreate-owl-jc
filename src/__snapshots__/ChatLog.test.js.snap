// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatLog Component tests renders the expected events in log 1`] = `
<div>
  <div>
    <div>
      Axe Kubicka
       -- 
      Hello world
       -- 
      7:50:46 AM
    </div>
  </div>
  <div>
    Caty Flucker gave a thumbs up 👍
  </div>
  <div>
    Frank Runciman gave a thumbs down 👎
  </div>
</div>
`;
